### PR TITLE
Dynamic On-Demand vSphere Slave

### PR DESCRIPTION
**Pull request by ghost1874**
This is an initial design for configuring jenkins to dynamically create vsphere clones to use as nodes.   During "Configure Jenkins" you can now add slave templates that define how new clones can be provisioned when a job needs a node and no node is able to accept the job.

### CODE REVIEW

**Comment by jswager** [comment]
Pretty impressive work.  @lfeagan - are you concerns addressed such that this code could be merged?

File: src/main/java/org/jenkinsci/plugins/vSphereCloudSlave.java [hunk 1]
```diff
@@ -113,9 +152,9 @@ public boolean isLaunchSupportForced() {
         return ((vSphereCloudLauncher) getLauncher()).getOverrideLaunchSupported() == Boolean.TRUE;
     }
 
-    public void setLaunchSupportForced(boolean slaveLaunchesOnBootup) {
-        ((vSphereCloudLauncher) getLauncher()).setOverrideLaunchSupported(slaveLaunchesOnBootup ? Boolean.TRUE : null);
-    }
+//    public void setLaunchSupportForced(boolean slaveLaunchesOnBootup) {
```
Comment (lfeagan) [inline src/main/java/org/jenkinsci/plugins/vSphereCloudSlave.java]: Perhaps remove this unused comment block. There are a few others.

**Comment by lfeagan** [comment]
@jswager My concerns have been addressed.

**Comment by jswager** [comment]
@ghost1874 - This PR will need to be udpated to handle the merges from previous PRs.

**Comment by ghost1874** [comment]
I looked at the build output.  I think it is still complaining on VSphere. java.   I have pulled that from jenkinsci/vsphere-cloud-plugin, accepted my changes, committed the changes, and pushed those changes back to ghost1874/vsphere-cloud-plugin.  If there is something else I need to do in-order to get this build fixed, please let me know.

**Pull request merged**